app/src/main/java/de/tum/in/tumcampusapp/managers/CafeteriaMenuManager.java

```diff
@@ -218,5 +220,60 @@ private void replaceIntoDb(CafeteriaMenu c) {
                 new String[]{String.valueOf(c.cafeteriaId),
                         Utils.getDateString(c.date), c.typeShort, c.typeLong,
                         String.valueOf(c.typeNr), c.name});
+        notifyFavoriteFoodService();
+    }
+
+    /**
+     * This method has to be called after we updated the local favorite database or if the cafeterias'
+     * menus have changed. It first checks whether there are any cafeterias, which serve a favorite food
+     * and only if there's at least one we query the location manager to see if it is the prefered one.
+     * Then a background service is started, which sends a notification at a given time and makes sure,
+     * that there's only one of these notifications per day.
+     */
+    public void notifyFavoriteFoodService(){
```

Comment:
This function is quite long, but does not have a single comment on what each of the blocks inside are doing. Please add some comments :+1: